src/main/kotlin/com/hj/leetcode/kotlin/problem1396/Solution.kt

```diff
@@ -0,0 +1,56 @@
+package com.hj.leetcode.kotlin.problem1396
+
+/**
+ * LeetCode page: [1396. Design Underground System](https://leetcode.com/problems/design-underground-system/);
+ */
+class UndergroundSystem() {
+
+    private val activeCheckIn = hashMapOf<CustomerID, CheckInRecord>()
+    private val tripTimeSummary = hashMapOf<Trip, TimeSummary>()
+
+    private data class CustomerID(val value: Int)
+
+    private data class CheckInRecord(val stationName: String, val timeStamp: Int)
+
+    private data class Trip(val fromStation: String, val toStation: String)
+
+    private data class TimeSummary(val totalTimes: Int, val numTrips: Int) {
+
+        fun averageTravelTime(): Double {
+            return totalTimes.toDouble() / numTrips
+        }
+    }
+
+    /* Complexity of N calls:
+     * Time O(N) and Space O(N);
+     */
+    fun checkIn(id: Int, stationName: String, t: Int) {
+        check(!activeCheckIn.contains(CustomerID(id))) {
+            "The customer has already checked in but has not yet checked out."
+        }
+        activeCheckIn[CustomerID(id)] = CheckInRecord(stationName, t)
+    }
+
+    /* Complexity of N calls:
+     * Time O(N) and Space O(N);
+     */
+    fun checkOut(id: Int, stationName: String, t: Int) {
+        val checkInRecord = activeCheckIn[CustomerID(id)] ?: throw IllegalStateException()
```

Comment:
Add message to the `IllegalStateException`

Reply:
Resolved at https://github.com/hj-core/hj-leetcode-kotlin/pull/73/commits/edb8487baf8ef18886f89689c519a8208d1f0331